### DISCUSSION TITLE
Cannot Hide Ulauncher window on startup

### DISCUSSION THREAD

**Discussion by TimilsinaBimal** [Ulauncher/Ulauncher#1053] [Q&A]
### Environment

ubuntu 22.04

### Bug description

I have setup ulauncher to start on login. But every time I open the computer(login/start) the ulauncher window appears. I have tried `--hide-window` flag too but it also didn't work.
Is this a bug? or how can I fix this.
Thank you!

### Log output

```shell
DBus name already taken. Ulauncher is probably backgrounded. Did you mean `ulauncher-toggle`?
```

### Communication guidelines

- [X] I have read and agree with Ulauncher's communication guidelines and checked the troubleshooting page for self-help.

### Not an extension issue

- [X] This issue applies to the Ulauncher application, not a specific extension.

### Ulauncher v6

- [X] I have made sure the bug isn't already fixed in v6.

**Comment by friday** [comment]
This message means you are already running Ulauncher and trying to start another instance of it. Install Gnome Tweaks and remove the autostart entries. You must have created one manually. Then you have to enable autostart in Ulauncher settings again.

**Reply by TimilsinaBimal** [reply]
I tried as you suggested, but it still is showing ulauncher window when I start my computer.
Here is the startup entry created after I clicked on `Launch at Login`.
```env GDK_BACKEND=x11 /usr/bin/ulauncher --hide-window```

**Comment by TimilsinaBimal** [comment]
I solved it using `ulauncher-toggle` instead of `ulauncher` in the startup entry.
i.e. Replaced ```env GDK_BACKEND=x11 /usr/bin/ulauncher --hide-window```
with 
```env GDK_BACKEND=x11 ulauncer-toggle --hide-window```
But I don't know if this is the right way.

**Reply by friday** [reply]
ulauncher-toggle doesn't do anything unless Ulauncher is running. All this does is that you're sending a call to toggle, but there is no app listening to the call. You might as well remove Ulauncher from autostart completely because you already have it starting some other way.

**Reply by TimilsinaBimal** [reply]
How can I completely remove it from the startup app? I have deleted the entry from gnome-tweaks startup-app list. Do I have to remove it from other places too?

**Reply by friday** [reply]
Press the cog wheel button to the right side of the text input to show the Ulauncher preferences. That's the only official way to enable/disable startup for Ulauncher. Anything you have done outside of that scope is up to you to know, understand and if necessary undo.

**Reply by friday** [reply]
And in Ulauncher v5 that's implemented to use the same type of implementation as gnome tweaks does btw. So if you removed it there that should be enough.

**Reply by TimilsinaBimal** [reply]
I did remove it but it still starts on startup. I think there is some entry in my startup but I didn't find it. which is why it is working as intended even after removing startup entry from gnome-tweaks.
